integration-tests/helpers/start-server-with-environment.js

```diff
@@ -0,0 +1,28 @@
+const setupServer = require("./setup-server")
+const { initDb } = require("./use-db")
+
+const startServerWithEnvironment = async ({ cwd, verbose, env }) => {
+  if (env) {
```

Comment:
we cant since it doens't reset the keys properly. In order to reset Node asks that you use `delete`
https://github.com/medusajs/medusa/pull/1768#discussion_r912450656

Reply:
That was because it was the reference to the env that was assigned and not a copy of the env stored in a new object, therefore, any changes on process.env was also applied on OLD_ENV, could you just try it out?

Reply:
If you test that 
```js
const processEnv = { ...process.env }
process.env.NEW_VALUE = "new value"
console.log(process.env.NEW_VALUE)
process.env = processEnv
console.log(process.env.NEW_VALUE)
```
everything works properly, in the other hand, if you do that, it can't work
```js
const processEnv = process.env
process.env.NEW_VALUE = "new value"
console.log(process.env.NEW_VALUE)
process.env = processEnv
console.log(process.env.NEW_VALUE)
```